src/transformers/models/roberta/modular_roberta.py

```diff
@@ -0,0 +1,777 @@
+
+        position_ids = torch.arange(
+            self.padding_idx + 1, sequence_length + self.padding_idx + 1, dtype=torch.long, device=inputs_embeds.device
+        )
+        return position_ids.unsqueeze(0).expand(input_shape)
+
+
+class RobertaSelfAttention(BertSelfAttention):
+    pass
+
+
+class RobertaSdpaSelfAttention(RobertaSelfAttention, BertSdpaSelfAttention):
+    pass
+
+
+class RobertaSelfOutput(BertSelfOutput):
+    pass
+
+
+ROBERTA_SELF_ATTENTION_CLASSES = {
+    "eager": RobertaSelfAttention,
+    "sdpa": RobertaSdpaSelfAttention,
+}
+
+
+class RobertaAttention(BertAttention):
+    def __init__(self, config, position_embedding_type=None):
+        super().__init__(config, position_embedding_type)
+        self.output = RobertaSelfOutput(config)
+
+
+class RobertaIntermediate(BertIntermediate):
+    pass
+
+
+class RobertaOutput(BertOutput):
+    pass
+
+
+class RobertaLayer(BertLayer):
+    def __init__(self, config):
+        super().__init__(config)
+        self.attention = RobertaAttention(config)
+        self.intermediate = RobertaIntermediate(config)
+        self.output = RobertaOutput(config)
+
+
+class RobertaEncoder(BertEncoder):
+    def __init__(self, config):
+        super().__init__(config)
+        self.layer = nn.ModuleList([RobertaLayer(config) for _ in range(config.num_hidden_layers)])
+
+
+class RobertaPooler(BertPooler):
+    pass
+
+
+class RobertaPreTrainedModel(PreTrainedModel):
+    """
+    An abstract class to handle weights initialization and a simple interface for downloading and loading pretrained
+    models.
+    """
+
+    config_class = RobertaConfig
+    base_model_prefix = "roberta"
+    supports_gradient_checkpointing = True
+    _no_split_modules = ["RobertaEmbeddings", "RobertaSelfAttention", "RobertaSdpaSelfAttention"]
+    _supports_sdpa = True
+
+    def _init_weights(self, module):
+        """Initialize the weights"""
+        if isinstance(module, nn.Linear):
+            # Slightly different from the TF version which uses truncated_normal for initialization
+            # cf https://github.com/pytorch/pytorch/pull/5617
+            module.weight.data.normal_(mean=0.0, std=self.config.initializer_range)
+            if module.bias is not None:
+                module.bias.data.zero_()
+        elif isinstance(module, nn.Embedding):
+            module.weight.data.normal_(mean=0.0, std=self.config.initializer_range)
+            if module.padding_idx is not None:
+                module.weight.data[module.padding_idx].zero_()
+        elif isinstance(module, nn.LayerNorm):
+            module.bias.data.zero_()
+            module.weight.data.fill_(1.0)
+
+
+class RobertaModel(RobertaPreTrainedModel, BertModel):
+    def __init__(self, config, add_pooling_layer=True):
+        super().__init__(config)
+        self.embeddings = RobertaEmbeddings(config)
+        self.encoder = RobertaEncoder(config)
+        self.pooler = RobertaPooler(config) if add_pooling_layer else None
+
```

Comment:
```suggestion
class RobertaModel(RobertaPreTrainedModel, BertModel):
    def __init__(self, config, add_pooling_layer=True):
        super().__init__(config)
        self.embeddings = RobertaEmbeddings(config)
```
this should be the only thing you have to write exlicitly